dts/arm/st/g4/stm32g474.dtsi

```diff
@@ -7,7 +7,19 @@
 #include <st/g4/stm32g473.dtsi>
 
 / {
+	aliases {
+		crc = &crc;
+	};
+
 	soc {
 		compatible = "st,stm32g474", "st,stm32g4", "simple-bus";
+
+		crc: crc@40023000 {
+			compatible = "st,stm32-crc";
+			reg = <0x40023000 0xA0>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB1 0x1000>;
+
+			status = "okay";
```

Comment:
Devices should be enabled only in boards.
So please move status activation in matching nucleo board.
Also please mention `crc` support in board's yaml file. Then, in test and sample yaml files add `depends_on: crc`.
Then, you can remove `platform_(allow|exclude)` fields.